drivers/src/sha2_512_384.rs

```diff
@@ -94,18 +174,19 @@ impl Sha384 {
                     // PANIC-FREE: Use buf.get() instead if buf[] as the compiler
                     // cannot reason about `offset` parameter to optimize out
                     // the panic call.
-                    if let Some(slice) = buf.get(offset..offset + SHA384_BLOCK_BYTE_SIZE) {
-                        let block = <&[u8; SHA384_BLOCK_BYTE_SIZE]>::try_from(slice).unwrap();
-                        self.digest_block(block, first, false)?;
-                        bytes_remaining -= SHA384_BLOCK_BYTE_SIZE;
+                    if let Some(slice) = buf.get(offset..offset + SHA2_BLOCK_BYTE_SIZE) {
+                        let block = <&[u8; SHA2_BLOCK_BYTE_SIZE]>::try_from(slice).unwrap();
+                        self.digest_block(ShaMode::Sha512, block, first, false)?;
+                        bytes_remaining -= SHA2_BLOCK_BYTE_SIZE;
```

Comment:
`sha512_digest` and `sha384_digest` are very similar functions, they just have different return types and different modes. Can have a single core implementation to avoid duplicating all this code?

Reply:
Probably can do it similarly to how you parameterized the Sha2DigestOp type.

Reply:
Created a helper function to separate the common code.